resolwe/flow/managers/base.py

```diff
@@ -53,37 +52,31 @@ def default(self, o):  # pylint: disable=method-hidden
 def dependency_status(data):
     """Return abstracted satus of dependencies.
 
-    STATUS_ERROR .. one dependency has error status
+    STATUS_ERROR .. one dependency has error status or was deleted
     STATUS_DONE .. all dependencies have done status
     None .. other
 
     """
-    for field_schema, fields in iterate_fields(data.input, data.process.input_schema):
-        if (field_schema['type'].lower().startswith('data:') or
-                field_schema['type'].lower().startswith('list:data:')):
-            name = field_schema['name']
-            value = fields[name]
+    parents_status = set(
+        DataDependency.objects.filter(
+            child=data, kind=DataDependency.KIND_IO
+        ).values_list('parent__status', flat=True)
```

Comment:
Since you are creating a set in Python, you could use distinct to reduce the number of entries on the database server?

Reply:
Good point, i will fix it.